src/Modularity.Core/Loaders/FileModuleLoader.cs

```diff
@@ -51,43 +53,57 @@ private IEnumerable<IModule> EnumerateAssembliesByConfig(string directory, strin
+                    EntryObjects = new List<IEntryObject>()
+                };
 
-                    foreach (var file in files)
+                try
+                {
+                    if (config.Files != null && config.Files.Length > 0)
+                        foreach (var file in config.Files)
+                            try
+                            {
+                                using (var ms = new MemoryStream(File.ReadAllBytes(file)))
+                                    AssemblyLoadContext.Default.LoadFromStream(ms);
+                            }
+                            catch (Exception ex)
+                            {
+                                Debug.WriteLine(ex.ToString());
+                                module.Exceptions?.Add(ex);
+                            }
+
+                    var mainAssemblyName = Path.Combine(directory, config.Name, config.Name + ".dll");
+
+                    foreach (var file in Directory.GetFiles(Path.Combine(directory, config.Name), "*.dll"))
                     {
+                        var isMainAssembly = file == mainAssemblyName;
+                        if (!config.LoadAllDependencies && !isMainAssembly)
+                            continue;
 
                         var bytes = File.ReadAllBytes(file);
-                        var assembly = Assembly.Load(bytes);
 
-                        module.Assemblies.Add(assembly);
-
-                        var entryObjectTypes = assembly.GetTypes().Where(x => x.IsClass && typeof(IEntryObject).IsAssignableFrom(x)).ToList();
-                        if (entryObjectTypes != null && entryObjectTypes.Count > 0)
+                        if (isMainAssembly)
                         {
-                            foreach (var entryobjectType in entryObjectTypes)
-                                module.EntryObjects.Add((IEntryObject)Activator.CreateInstance(entryobjectType));
-
-                            module.EntryObject = (IEntryObject)Activator.CreateInstance(entryObjectTypes.FirstOrDefault());
+                            var assembly = Assembly.Load(bytes);
                             module.Assembly = assembly;
-                            if (!config.LoadAllDependencies)
-                                break;
+                            module.AssemblyName = assembly.FullName;
+                            module.EntryObjects = assembly.GetTypes().Where(x => typeof(IEntryObject).IsAssignableFrom(x)).Select(s => (IEntryObject)Activator.CreateInstance(s)).ToList();
+                            module.EntryObject = module.EntryObjects?.FirstOrDefault();
+                            module.Name = config.Name;
+                        }
+                        else
+                        {
+                            using (var ms = new MemoryStream(bytes))
+                            {
+                                var assembly = AssemblyLoadContext.Default.LoadFromStream(ms);
```

Comment:
Include assembly current AppDomain, and made it usable via using `AssemblyLoadContext` (ALC)